### PR TITLE
add update workflow endpoint

### PR DESCRIPTION
**Pull request by jackschofield23**
Changes required for #61  .

### Description

Add update workflow endpoint

### Status
**Ready**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [ ] Code coverage above 70%
- [ ] TODO items resolved

### CODE REVIEW

File: src/Common/Services/WorkflowService.cs [hunk 1]
```diff
@@ -32,5 +32,20 @@ public async Task<string> CreateAsync(Workflow workflow)
 
             return await _workflowRepository.CreateAsync(workflow);
         }
+
+        public async Task<string> UpdateAsync(Workflow workflow, string id)
```
Comment (lillie-dae) [inline src/Common/Services/WorkflowService.cs:36]: Maybe make this a nullable string?

File: tests/UnitTests/Common.Tests/Monai.Deploy.WorkflowManger.Common.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (lillie-dae) [inline tests/UnitTests/Common.Tests/Monai.Deploy.WorkflowManger.Common.Tests.csproj:1]: Header 🍡 🥇

**Review by lillie-dae — APPROVED** [review]

File: src/Common/Services/WorkflowService.cs (hunk 1 above)
Comment (jackschofield23) [reply src/Common/Services/WorkflowService.cs:38]: Added

File: tests/UnitTests/Common.Tests/Monai.Deploy.WorkflowManger.Common.Tests.csproj (hunk 2 above)
Comment (jackschofield23) [reply tests/UnitTests/Common.Tests/Monai.Deploy.WorkflowManger.Common.Tests.csproj:6]: Added 💯 🎱

File: src/WorkflowManager/Controllers/WorkflowsController.cs [hunk 3]
```diff
@@ -107,4 +107,46 @@ public async Task<IActionResult> CreateAsync([FromBody] Workflow workflow)
...
         }
     }
+
+    /// <summary>
+    /// Updates a workflow and creates a new revision
+    /// </summary>
+    /// <param name="workflow">The Workflow.</param>
+    /// <returns>The ID of the created Workflow.</returns>
+    [Route("{id}")]
+    [HttpPut]
```
Comment (mocsharp) [inline src/WorkflowManager/Controllers/WorkflowsController.cs:117]: I think these 2 lines can be combined into 1 as: `[HttpPut("{id}")]`

Also, do we want to decorate all APIs with: `Consumes`, `Produces`, `ProducesResponseType` attributes?

**Pull request merged**